From: Daniel Leidert (dale) <daniel.leidert@wgdd.de>
Subject: Fix some directory values, correct binary and translation install
 All installs are later done using dh_install. Respect the CFLAGS
 set in debian/rules.
 .
 Also fix #436782 and respect/handle nostrip build option (policy 10.1).
 See <URL:http://bugs.debian.org/436782>.

diff -urNad easychem~/Makefile.linux easychem/Makefile.linux
--- easychem~/Makefile.linux	2005-05-04 10:10:07.000000000 +0200
+++ easychem/Makefile.linux	2007-12-18 01:43:14.000000000 +0100
@@ -1,17 +1,18 @@
 # The C compiler
 CC=gcc
+C_FLAGS=
 
 # The following line must be uncommented if you want to specify a place
 # where the 'gs' (or 'pstoedit') program is.
 
-#C_FLAGS="$(C_FLAGS) -DGS_PATH=/opt/bin"
-#C_FLAGS="$(C_FLAGS) -DPSTOEDIT_PATH=/opt/bin"
+#C_FLAGS += -DGS_PATH=\"/usr/bin\" # don't enable this!
+#C_FLAGS += -DPSTOEDIT_PATH=\"/usr/bin\" # don't enable this!
 
 # Here we define the prefix where EasyChem will be installed
-PREFIX=/usr/local
+PREFIX=/usr
 
 # Optimisation and warning flags
-C_FLAGS=-Wall -W -Wpointer-arith -O2 -pedantic -std=c99 -DUNIX -DI18N -DPREFIX=\"$(PREFIX)\"
+C_FLAGS += -W -Wpointer-arith -pedantic -std=c99 -DUNIX -DI18N -DPREFIX=\"$(PREFIX)\"
 
 # GTK flags and librairies
 GTK_FLAGS=`pkg-config --cflags gtk+-2.0`
@@ -28,8 +29,8 @@
 all: easychem po/
 
 easychem: postscript/ graph/ $(OBJECTS)
-	$(CC) -o easychem $(OBJECTS) $(GTK_LIBS) $(C_FLAGS)
-	strip easychem
+	$(CC) $(LDFLAGS) -o easychem $(OBJECTS) $(GTK_LIBS) $(C_FLAGS) $(CFLAGS)
+	# strip easychem # don't strip, see bug #436782
 
 clean: FORCE
	-cd graph; make clean
@@ -38,21 +39,21 @@
	-rm $(OBJECTS) easychem
 
 export.o: export.c common.h bonds.h postscript/
-	$(CC) $(GTK_FLAGS) $(C_FLAGS) -c export.c
+	$(CC) $(GTK_FLAGS) $(C_FLAGS) $(CFLAGS) -c export.c
 bonds.o: bonds.c common.h bonds.h
-	$(CC) $(GTK_FLAGS) $(C_FLAGS) -c bonds.c
+	$(CC) $(GTK_FLAGS) $(C_FLAGS) $(CFLAGS) -c bonds.c
 easychem.o: easychem.c common.h bonds.h export.h library.h drawing.h
-	$(CC) $(GTK_FLAGS) $(C_FLAGS) -c easychem.c
+	$(CC) $(GTK_FLAGS) $(C_FLAGS) $(CFLAGS) -c easychem.c
 auxi.o: auxi.c common.h auxi.h
-	$(CC) $(GTK_FLAGS) $(C_FLAGS) -c auxi.c
+	$(CC) $(GTK_FLAGS) $(C_FLAGS) $(CFLAGS) -c auxi.c
 library.o: library.c common.h library.h
-	$(CC) $(GTK_FLAGS) $(C_FLAGS) -c library.c
+	$(CC) $(GTK_FLAGS) $(C_FLAGS) $(CFLAGS) -c library.c
 drawing.o: drawing.c common.h drawing.h
-	$(CC) $(GTK_FLAGS) $(C_FLAGS) -c drawing.c
+	$(CC) $(GTK_FLAGS) $(C_FLAGS) $(CFLAGS) -c drawing.c
 dialogs.o: dialogs.c common.h dialogs.h graph/
-	$(CC) $(GTK_FLAGS) $(C_FLAGS) -c dialogs.c
+	$(CC) $(GTK_FLAGS) $(C_FLAGS) $(CFLAGS) -c dialogs.c
 detect.o: detect.c common.h detect.h
-	$(CC) $(GTK_FLAGS) $(C_FLAGS) -c detect.c
+	$(CC) $(GTK_FLAGS) $(C_FLAGS) $(CFLAGS) -c detect.c
 
 # The rules for subdirectories
 graph/: FORCE
diff -urNad easychem~/po/Makefile easychem/po/Makefile
--- easychem~/po/Makefile	2005-05-04 10:10:07.000000000 +0200
+++ easychem/po/Makefile	2007-12-18 01:41:57.000000000 +0100
@@ -1,7 +1,8 @@
-all: fr.mo
+all: fr/LC_MESSAGES/easychem.mo
 
-%.mo: %.po
-	msgfmt -o $*.mo $*.po 
+%/LC_MESSAGES/easychem.mo: %.po
+	mkdir -p $*/LC_MESSAGES
+	msgfmt -o $*/LC_MESSAGES/easychem.mo $*.po 
 
 clean:
-	-rm *.mo
+	-rm -rf fr
